src/devices/src/virtio/balloon/device.rs

```diff
@@ -385,14 +382,9 @@ impl Balloon {
     }
 
     pub(crate) fn signal_used_queue(&self) -> Result<(), BalloonError> {
-        self.interrupt_status
-            .fetch_or(VIRTIO_MMIO_INT_VRING as usize, Ordering::SeqCst);
-
-        self.interrupt_evt.write(1).map_err(|e| {
-            error!("Failed to signal used queue: {:?}", e);
-            BalloonError::FailedSignalingUsedQueue(e)
-        })?;
-        Ok(())
+        self.irq_trigger
+            .trigger_irq(IrqType::Vring)
+            .map_err(BalloonError::InterruptError)
```

Comment:
nit: consistency: some devices increment `event_fail` metric for failed irqs while others don't.

Reply:
Fixed this for the balloon. The vsock device doesn't even have this metric.